merkato/utils/__init__.py

```diff
@@ -71,3 +71,29 @@ def generate_complete_merkato_configs(merkato_tuples):
 		merkato_complete_configs.append(complete_config)
 
 	return merkato_complete_configs
+
+def get_allocated_pair_balances(exchange, base, coin):
+    allocated_pair_balances = {
+        'base': 0,
+        'coin': 0
+    }
+
+    merkatos = get_merkatos_by_exchange(exchange)
+    for merkato in merkatos:
+        if merkato['base'] == base:
+            allocated_pair_balances['base'] += merkato['bid_reserved_balance']
+
+        if merkato['alt'] == coin:
+            allocated_pair_balances['coin'] += merkato['ask_reserved_balance']
+    return allocated_pair_balances
+
+def check_reserve_balances(total_balances, allocated_balances, coin_reserve, base_reserve):
+    remaining_balances = {
+        'base': total_balances['base']['amount'] - allocated_balances['base'],
+        'coin': total_balances['coin']['amount'] - allocated_balances['coin']
+    }
+    if remaining_balances['base'] < base_reserve:
+        raise ValueError('Cannot create merkato, the suggested base reserve will exceed the amount of the base asset on the exchange.')
```

Comment:
Why is the 'check reserve balances' function aware of the 'create merkatos' function?

Reply:
in terms of it's error message

Reply:
the only time its called is in the init of the merkato. it fails and stops merkato creation from happening

Reply:
The function doesn't have a return value, or modify any variables. Can you restructure it such that it returns a truthy value, and the parent function raises the error in its own context? I'm not comfortable with this util function being context aware of the function that uses it.